### PR TITLE
Add parsing new section headings in CODEOWNERS files

### PR DESCRIPTION
**Pull request by Totoo98**
Hi! My team would like to be able to parse new section headings in CODEOWNERS files, namely the required reviewer count and the default owners (https://docs.gitlab.com/ee/user/project/codeowners/reference.html#section-headings).

This change improves the parsing of section lines, and adds a few tests for section heading parsing as well as a test for files with lines ending with CRLF.

Please note there is a breaking change as the CodeOwnersSection constructor now takes an int instead of bool, since we can now deduce IsOptional from RequiredReviewerCount.
I couldn't find a breaking change guidelines for this project, but if we don't want one right now I suggest keeping both constructors, set the RequiredReviewerCount at 1 if isOptional is false in the old one, and ideally mark it as deprecated.

Thanks ⁠😊
Thomas

### CODE REVIEW

File: src/Meziantou.Framework.CodeOwners/CodeOwnersSection.cs [hunk 1]
```diff
@@ -5,14 +5,21 @@ namespace Meziantou.Framework.CodeOwners;
 [StructLayout(LayoutKind.Auto)]
 public readonly struct CodeOwnersSection : IEquatable<CodeOwnersSection>
 {
-    public CodeOwnersSection(string name, bool isOptional)
+    public CodeOwnersSection(string name, int requiredReviewerCount = 1, IReadOnlyCollection<string> defaultOwners = null)
```
Comment (meziantou) [inline src/Meziantou.Framework.CodeOwners/CodeOwnersSection.cs]: Can you make it `internal`? I don't think it need to be public.

File: tests/Meziantou.Framework.CodeOwners.Tests/CodeOwnersParserTests.cs [hunk 2]
```diff
@@ -180,4 +180,86 @@ public void ParseCodeOwnersWithPatternsWithoutMembers()
...
+
+        var expected = new CodeOwnersEntry[]
+        {
+            CodeOwnersEntry.FromUsername(0, "*", "user1", section: new CodeOwnersSection("Test1")),
+            CodeOwnersEntry.FromUsername(1, "*", "user2", section: new CodeOwnersSection("Test2", 2)),
+            CodeOwnersEntry.FromUsername(2, "*", "user3", section: new CodeOwnersSection("Test3", 1, ["@defaultOwner"])),
+        };
+
+        actual.Should().Equal(expected);
+    }
```
Comment (meziantou) [inline tests/Meziantou.Framework.CodeOwners.Tests/CodeOwnersParserTests.cs:374]: What's the behavior of GitLab in the following cases? Can you add test that matches the GitLab bahvior?
- `^[section][2]` (From the doc "Optional sections ignore the number of approvals required")
- `[section] [2]`
- `[section] # [2]`

**Comment by meziantou** [comment]
Can you bump the version in the csproj, so it will deploy a new version after merging this PR?

File: src/Meziantou.Framework.CodeOwners/CodeOwnersSection.cs (hunk 1 above)
Comment (Totoo98) [reply src/Meziantou.Framework.CodeOwners/CodeOwnersSection.cs]: Sure, addressed 👌

File: tests/Meziantou.Framework.CodeOwners.Tests/CodeOwnersParserTests.cs (hunk 2 above)
Comment (Totoo98) [reply tests/Meziantou.Framework.CodeOwners.Tests/CodeOwnersParserTests.cs:374]: Good catch, I'll have a look next week into those edge cases :)

Comment (Totoo98) [reply tests/Meziantou.Framework.CodeOwners.Tests/CodeOwnersParserTests.cs:374]: I've updated the parser and tests according to how GitLab handle those cases, as well as a few more similar ambiguities.
